pkg/resources/helpers.go

```diff
@@ -142,3 +146,77 @@ func IsDataType() schema.SchemaValidateFunc { //nolint:staticcheck
 		return warnings, errors
 	}
 }
+
+// IsValidIdentifier is a validator that can be used for validating identifiers passed in resources and data sources.
+// Typically, we expect passed identifiers to be a variation of sdk.ObjectIdentifier. To use this function, pass it as
+// a validation function on identifier field with generic parameter set to the desired sdk.ObjectIdentifier implementation.
+func IsValidIdentifier[T sdk.ObjectIdentifier]() schema.SchemaValidateDiagFunc {
+	return func(value any, path cty.Path) diag.Diagnostics {
+		var diags diag.Diagnostics
+
+		// For now, we won't support sdk.ExternalObjectIdentifiers. The reason behind it is that the functions that parse identifiers are not
+		// able to differentiate between sdk.ExternalObjectIdentifiers and sdk.DatabaseObjectIdentifier or sdk.SchemaObjectIdentifier,
+		// because sdk.ExternalObjectIdentifiers has varying parts count (2 or 3).
+		if _, ok := any(sdk.ExternalObjectIdentifier{}).(T); ok {
```

Comment:
we can replace `T sdk.ObjectIdentifier` with `[T sdk.AccountObjectIdentifier | sdk.DatabaseObjectIdentifier | sdk.SchemaObjectIdentifier | sdk.TableColumnIdentifier]` and we don't have to handle this case then

Reply:
True, didn't think of that